app/src/main/java/com/example/parkingsystem/mvp/model/ReservationModel.java

```diff
@@ -1,29 +1,81 @@
 package com.example.parkingsystem.mvp.model;
 
 import com.example.parkingsystem.mvp.contract.ReservationContract;
+import com.example.parkingsystem.mvp.model.reservation.Reservation;
+import com.example.parkingsystem.mvp.model.reservation.ReservationInfomation;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
 
 public class ReservationModel implements ReservationContract.ReservationModelContract {
 
-    private String date;
-    private String time;
+    private static final String BUTTON_START_DATE = "Start date";
+    private static final String FORMAT_DATE = "dd/MM/yyyy hh:mm a";
+    private static final String START_DATE = "Start date: ";
+    private static final String FINISH_DATE = "Finish date: ";
+    private static final String SECURITY_CODE = "Security code: ";
+    private static final String PLACE = "Place: ";
+    private static final String LINE_JUMP = "\n";
+
+    private ReservationInfomation reservationInfomation;
+    private String buttonText;
+    private Calendar startDateAndTime = null;
+    private Calendar finishDateAndTime = null;
+    private boolean startTime = false;
+    private boolean finishTime = false;
 
     @Override
-    public void saveDate(String date) {
-        this.date = date;
+    public void saveDate(int year, int month, int dayOfMonth) {
+        if (buttonText.equals(BUTTON_START_DATE)) {
```

Comment:
you shouldnt use this metod to distinguish start for end button.

if you think in the future to change the text of the btn, you will have a dependence in your logic.

i think you can use a flag in the model, or a != empty logic for this case

Reply:
done